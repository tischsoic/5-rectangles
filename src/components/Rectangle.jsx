import React from 'react';
import PropTypes from 'prop-types';

import './Rectangle.scss';

const Rectangle = ({ id, top, left, width, height, onRemove }) => {
  const style = {
    top,
    left,
    width,
    height,
  };

  /* eslint-disable jsx-a11y/click-events-have-key-events */
  return (
    <div
      onClick={(event) => {
        if (event.altKey) {
          event.stopPropagation();
          onRemove(id);
        }
      }}
      role="presentation"
      className="rectangle"
      style={style}
    />
  );
  /* eslint-enable */
};

Rectangle.propTypes = {
  id: PropTypes.number.isRequired,
  top: PropTypes.number.isRequired,
  left: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  onRemove: PropTypes.func.isRequired,
};

export default Rectangle;
