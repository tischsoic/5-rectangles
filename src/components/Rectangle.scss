@import '../index.scss';

.rectangle {
  position: absolute;
  background-color: $rectangle-bgcolor;
}

@media print {
  .rectangle {
    background-color: $body-bgcolor;
  }
}
