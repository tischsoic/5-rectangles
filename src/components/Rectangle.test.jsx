import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Rectangle from './Rectangle';

describe('Rectangle', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Rectangle
          key={0}
          id={0}
          top={10}
          left={20}
          width={30}
          height={40}
          onRemove={() => {}}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('calls onRemove with rectangle ID after click with ALT key', () => {
    const onRemoveMock = jest.fn();
    const rectangleId = 123;
    const rectangle = shallow(
      <Rectangle
        key={0}
        id={rectangleId}
        top={10}
        left={20}
        width={30}
        height={40}
        onRemove={onRemoveMock}
      />,
    );

    rectangle.find('div').simulate('click', {
      altKey: true,
      stopPropagation: () => {},
    });
    expect(onRemoveMock.mock.calls.length).toEqual(1);
    expect(onRemoveMock.mock.calls[0][0]).toEqual(rectangleId);
  });
});
