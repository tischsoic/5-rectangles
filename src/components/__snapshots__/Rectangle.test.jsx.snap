// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rectangle renders correctly 1`] = `
<div
  className="rectangle"
  onClick={[Function]}
  role="presentation"
  style={
    Object {
      "height": 40,
      "left": 20,
      "top": 10,
      "width": 30,
    }
  }
/>
`;
